### ISSUE TITLE
CA1822 false positive for partial methods with attributes

### ISSUE THREAD

**Issue by aelij** [dotnet/roslyn-analyzers#4733]
### Analyzer

**Diagnostic ID**: [CA1822](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1822): Mark members as static

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.102](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

CA1822 is falsely detected on partial methods that have attributes (it appears as soon as I type `[]` before a method) even if they access instance members.

### Steps To Reproduce

```cs
    partial class C
    {
        [Description]
        partial void M(); // CA1822
    }

    partial class C
    {
        private int _i;

        partial void M()
        {
            ++_i;
        }
    }
```

### Expected behavior

No CA1822

### Actual behavior

```
Severity	Code	Description	Project	File	Line	Suppression State
Warning	CA1822	Member 'M' does not access instance data and can be marked as static	SampleLogger	Program.cs	40	Active
```

**Comment by mavasani** [comment]
This could either be a Roslyn bug in analyzer execution on source generated files OR may need a fix in the analyzer.

**Comment by mavasani** [comment]
One thing to check would be any behavior difference in command line build case and IDE intellisense case.

**Comment by aelij** [comment]
If I set `CA1822` to a warning, I see it in `dotnet build`, so not an IDE issue.

**Comment by aelij** [comment]
My mistake - it's unrelated to generators. Just a bug in the analyzer:

```cs
    partial class C
    {
        [Description]
        partial void M(); // CA1822
    }

    partial class C
    {
        private int _i;

        partial void M()
        {
            ++_i;
        }
    }
```

**Comment by mjameson-se** [comment]
I'm also seeing CA1801 in this same scenario if the method has a parameter. Super annoying as I'm trying to write a source generator which generates code to implement my partial method based on the presence of an attribute. Looks like I can suppress the warning by adding the suppression attributes to the generated code though.
